### ISSUE TITLE
Fails on projects that have a directory with no read access

### ISSUE THREAD

**Issue by p-himik** [jpmonettas/clograms#39]
One of my projects has a `.data/var/lib/postgresql/data` directory that's only readable for the user that runs a PostgreSQL container with Podman.

Running Clograms on that project fails with
```
Indexing  . for platform  :clj watching: false
Execution error (AccessDeniedException) at sun.nio.fs.UnixException/translateToIOException (UnixException.java:90).
./.data/var/lib/postgresql/data
```

<details>
<summary>The full exception map</summary>

```clj
{:clojure.main/message
 "Execution error (AccessDeniedException) at sun.nio.fs.UnixException/translateToIOException (UnixException.java:90).\n./.data/var/lib/postgresql/data\n",
 :clojure.main/triage
 {:clojure.error/class java.nio.file.AccessDeniedException,
  :clojure.error/line 90,
  :clojure.error/cause "./.data/var/lib/postgresql/data",
  :clojure.error/symbol
  sun.nio.fs.UnixException/translateToIOException,
  :clojure.error/source "UnixException.java",
  :clojure.error/phase :execution},
 :clojure.main/trace
 {:via
  [{:type java.io.UncheckedIOException,
    :message
    "java.nio.file.AccessDeniedException: ./.data/var/lib/postgresql/data",
    :at
    [java.nio.file.FileTreeIterator
     fetchNextIfNeeded
     "FileTreeIterator.java"
     87]}
   {:type java.nio.file.AccessDeniedException,
    :message "./.data/var/lib/postgresql/data",
    :at
    [sun.nio.fs.UnixException
     translateToIOException
     "UnixException.java"
     90]}],
  :trace
  [[sun.nio.fs.UnixException
    translateToIOException
    "UnixException.java"
    90]
   [sun.nio.fs.UnixException asIOException "UnixException.java" 115]
   [sun.nio.fs.UnixFileSystemProvider
    newDirectoryStream
    "UnixFileSystemProvider.java"
    477]
   [java.nio.file.Files newDirectoryStream "Files.java" 481]
   [java.nio.file.FileTreeWalker visit "FileTreeWalker.java" 301]
   [java.nio.file.FileTreeWalker next "FileTreeWalker.java" 374]
   [java.nio.file.FileTreeIterator
    fetchNextIfNeeded
    "FileTreeIterator.java"
    83]
   [java.nio.file.FileTreeIterator hasNext "FileTreeIterator.java" 103]
   [java.util.Spliterators$IteratorSpliterator
    tryAdvance
    "Spliterators.java"
    1949]
   [java.util.stream.StreamSpliterators$WrappingSpliterator
    lambda$initPartialTraversalState$0
    "StreamSpliterators.java"
    292]
   [java.util.stream.StreamSpliterators$AbstractWrappingSpliterator
    fillBuffer
    "StreamSpliterators.java"
    206]
   [java.util.stream.StreamSpliterators$AbstractWrappingSpliterator
    doAdvance
    "StreamSpliterators.java"
    169]
   [java.util.stream.StreamSpliterators$WrappingSpliterator
    tryAdvance
    "StreamSpliterators.java"
    298]
   [java.util.Spliterators$1Adapter hasNext "Spliterators.java" 681]
   [clojure.lang.RT$4 invoke "RT.java" 522]
   [clojure.lang.LazySeq sval "LazySeq.java" 42]
   [clojure.lang.LazySeq seq "LazySeq.java" 51]
   [clojure.lang.RT seq "RT.java" 535]
   [clojure.core$seq__5467 invokeStatic "core.clj" 139]
   [clojure.core$filter$fn__5962 invoke "core.clj" 2826]
   [clojure.lang.LazySeq sval "LazySeq.java" 42]
   [clojure.lang.LazySeq seq "LazySeq.java" 58]
   [clojure.lang.RT seq "RT.java" 535]
   [clojure.core$seq__5467 invokeStatic "core.clj" 139]
   [clojure.core.protocols$seq_reduce invokeStatic "protocols.clj" 24]
   [clojure.core.protocols$fn__8236 invokeStatic "protocols.clj" 75]
   [clojure.core.protocols$fn__8236 invoke "protocols.clj" 75]
   [clojure.core.protocols$fn__8178$G__8173__8191
    invoke
    "protocols.clj"
    13]
   [clojure.core$reduce invokeStatic "core.clj" 6887]
   [clojure.core$mapv invokeStatic "core.clj" 6971]
   [clojure.core$mapv invoke "core.clj" 6971]
   [clindex.utils$all_files invokeStatic "utils.clj" 34]
   [clindex.utils$all_files invoke "utils.clj" 30]
   [clindex.scanner$find_project_in_dir invokeStatic "scanner.clj" 60]
   [clindex.scanner$find_project_in_dir invoke "scanner.clj" 57]
   [clindex.api$index_project_BANG_ invokeStatic "api.clj" 156]
   [clindex.api$index_project_BANG_ invoke "api.clj" 145]
   [clograms.core$re_index_all invokeStatic "core.clj" 28]
   [clograms.core$re_index_all invoke "core.clj" 25]
   [clograms.server$_main invokeStatic "server.clj" 74]
   [clograms.server$_main doInvoke "server.clj" 46]
   [clojure.lang.RestFn applyTo "RestFn.java" 137]
   [clojure.lang.Var applyTo "Var.java" 705]
   [clojure.core$apply invokeStatic "core.clj" 667]
   [clojure.main$main_opt invokeStatic "main.clj" 514]
   [clojure.main$main_opt invoke "main.clj" 510]
   [clojure.main$main invokeStatic "main.clj" 664]
   [clojure.main$main doInvoke "main.clj" 616]
   [clojure.lang.RestFn applyTo "RestFn.java" 137]
   [clojure.lang.Var applyTo "Var.java" 705]
   [clojure.main main "main.java" 40]],
  :cause "./.data/var/lib/postgresql/data"}}
```

</details>

**Comment by p-himik** [comment]
Some people online say that `Files/walk` is pretty much a no-go when you don't fully control the file tree and can guarantee its state. And that a working alternative is a bit more cumbersome `Files/walkFileTree` that requires a visitor: https://stackoverflow.com/a/41542026

But I just took a peek at the source code of Clindex and it seems that `Files/walk` can easily be replaced with the `file-seq` function built into Clojure. It doesn't fail when there's no access to any of the files/dirs.

**Comment by jpmonettas** [comment]
Oh nice, will give it a try on Monday. Thanks!

**Comment by jpmonettas** [comment]
This is fixed on 0.1.148!
